internal/services/cdn/cdn_frontdoor_rule_resource.go

```diff
@@ -0,0 +1,1276 @@
+package cdn
+
+import (
+	"fmt"
+	"time"
+
+	cdn "github.com/Azure/azure-sdk-for-go/services/cdn/mgmt/2021-06-01/cdn"
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/tf"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	cdnFrontDoorRuleActions "github.com/hashicorp/terraform-provider-azurerm/internal/services/cdn/frontdoorruleactions"
+	cdnFrontDoorRuleConditions "github.com/hashicorp/terraform-provider-azurerm/internal/services/cdn/frontdoorruleconditions"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/cdn/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/cdn/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/timeouts"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+func resourceCdnFrontDoorRule() *pluginsdk.Resource {
+	return &pluginsdk.Resource{
+		Create: resourceCdnFrontDoorRuleCreate,
+		Read:   resourceCdnFrontDoorRuleRead,
+		Update: resourceCdnFrontDoorRuleUpdate,
+		Delete: resourceCdnFrontDoorRuleDelete,
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Create: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Read:   pluginsdk.DefaultTimeout(5 * time.Minute),
+			Update: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Delete: pluginsdk.DefaultTimeout(30 * time.Minute),
+		},
+
+		Importer: pluginsdk.ImporterValidatingResourceId(func(id string) error {
+			_, err := parse.FrontDoorRuleID(id)
+			return err
+		}),
+
+		Schema: map[string]*pluginsdk.Schema{
+
+			"name": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.CdnFrontDoorRuleName,
+			},
+
+			"cdn_frontdoor_rule_set_id": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.FrontDoorRuleSetID,
+			},
+
+			"behavior_on_match": {
+				Type:     pluginsdk.TypeString,
+				Optional: true,
+				Default:  string(cdn.MatchProcessingBehaviorContinue),
```

Comment:
I don't believe so since this is the same behavior that the API has which has been documented in the MS Docs site. I think changing the behavior between toolsets can be confusing to our end users.